### PR TITLE
Global Validation Unit Testing

### PR DESCRIPTION
**Pull request by jphechter**
## Purpose

Create a proof-of-concept approach for testing our data validation.

#### Linked Issues to Close

Oy2-XXXX

## Approach

Initially just wanted to take a stab at putting guard rails around some of our primary functionality. There's a lot that could be done here, but also wanted to get feedback from the team before digging in too deeply. Longer term, I would like data for these tests to be generated based off what comes out of our measures. The app is changing a lot and hard coded values will likely fall out of date quickly.

If we do decide to go this route, it would be great if any time we work on a validation function we also write accompanying unit tests so that this stays up to date with what we're building.

#### Pull Request Creator Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [ ] Someone has been assigned this PR.
- [ ] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

File: services/ui-src/src/measures/globalValidations/tests/_helper.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { getPerfMeasureRateArray } from "measures/globalValidations";
+
+export const test_setup = (data: DefaultFormData) => {
+  return {
+    ageGroups: exampleData.qualifiers!,
+    performanceMeasureArray: getPerfMeasureRateArray(data, exampleData),
+    OPM: data[DC.OPM_RATES],
+  };
+};
+
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/globalValidations/tests/_helper.ts:13]: Might be good to move this and the formData to src/utils/testUtils for consistency with other helper functionality

File: services/ui-src/src/measures/globalValidations/tests/_testFormData.ts [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+      ],
+      InitiationofAODTreatmentTotalAODAbuseorDependence: [
+        {
+          [DC.LABEL]: "Ages 18 to 64",
+          [DC.RATE]: "",
+          [DC.NUMERATOR]: "",
+          [DC.DENOMINATOR]: "",
+        },
+        {
+          [DC.LABEL]: "Age 65 and older",
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/globalValidations/tests/_testFormData.ts:308]: We could create sample qualifiers and categories above this to remove these stray strings

File: services/ui-src/src/measures/globalValidations/tests/ensureBothDatesCompletedInRange.test.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  2  │ Complete   │ Complete  │
+  3  │ Undefined  │ Undefined │
+  4  │ Empty      │ Empty     │
+  5  │ Empty      │ Undefined │
+  6  │ Undefined  │ Empty     │
+  7  │ 0          │ 0         │
+  8  │ Complete   │ Empty     │
+  9  │ Empty      │ Complete  │
+  10 │ Month      │ Year      │
+  11 │ Year       │ Month     │
```
Comment (daniel-belcher) [inline services/ui-src/src/measures/globalValidations/tests/ensureBothDatesCompletedInRange.test.ts:21]: Really like the breakdown of all cases

**Review by daniel-belcher — APPROVED** [review]

File: services/ui-src/src/measures/globalValidations/tests/ensureBothDatesCompletedInRange.test.ts (hunk 3 above)
Comment (benjaminpaige) [reply services/ui-src/src/measures/globalValidations/tests/ensureBothDatesCompletedInRange.test.ts:21]: yeah this is really great!

**Review by benjaminpaige — APPROVED** [review]
yes, love this! great job

**Pull request merged**